### PR TITLE
Code improvement and gitignore

### PR DESCRIPTION
**Pull request by luizkowalski**

### CODE REVIEW

File: lib/cloudwatchlogger/client/aws_sdk/threaded.rb [hunk 1]
```diff
@@ -33,9 +35,11 @@ def start_thread
...
 
       class DeliveryThread < Thread
-
-        def initialize(credentials, log_group_name, log_stream_name, opts={})
-          @credentials, @log_group_name, @log_stream_name, @opts = credentials, log_group_name, log_stream_name, opts
+        def initialize(credentials, log_group_name, log_stream_name, opts = {})
+          @credentials = credentials
+          @log_group_name = log_group_name
+          @log_stream_name = log_stream_name
+          @opts = opts
```
Comment (zshannon) [inline lib/cloudwatchlogger/client/aws_sdk/threaded.rb:44]: AFAIK, need to wait until we're done mutating `opts` (next two lines) before we store it on the class. Whether this is true or not, I'd like the code to read consistently so it's obvious we intend to be concerned with one object here.

**Review by zshannon — CHANGES_REQUESTED** [review]
Are you using an automated formatter? It'd be nice to include the formatting preferences you've got in a reproducible, consistent way. I'm okay with accepting arbitrary syntax formatting changes if they reduce the entropy of the project overall.

File: lib/cloudwatchlogger/client/aws_sdk/threaded.rb (hunk 1 above)
Comment (luizkowalski) [reply lib/cloudwatchlogger/client/aws_sdk/threaded.rb:44]: > need to wait until we're done mutating opts (next two lines) before we store it on the class

but the code was not doing this before anyway...this was just a parallel assignment that I broke down into multiple lines to better reading. If you think that `opts` should be mutate *before* assign I can move up the next two lines

**Comment by luizkowalski** [comment]
I'm not using any code formatter but I'm fine with adding something like `.editorconfig` though I would do it in another PR

**Comment by zshannon** [comment]
I was thinking something like [rubocop](https://rubocop.readthedocs.io/en/latest/) to enforce their [community style guide](https://github.com/rubocop-hq/ruby-style-guide). Are you game to set that up?

File: lib/cloudwatchlogger/client/aws_sdk/threaded.rb (hunk 1 above)
Comment (zshannon) [reply lib/cloudwatchlogger/client/aws_sdk/threaded.rb:44]: I do think `opts` should be mutated before assigning it off to the class.  Moving the mutation up is 👍

**Comment by luizkowalski** [comment]
yes, I'll set up Rubocop in another PR later this week

**Review by zshannon — APPROVED** [review]

**Pull request merged**